pkg/query/querier.go

```diff
@@ -360,6 +364,7 @@ func (q *querier) selectFn(ctx context.Context, hints *storage.SelectHints, ms .
 		SkipChunks:              q.skipChunks,
 		Step:                    hints.Step,
 		Range:                   hints.Range,
+		SortWithoutLabels:       replicaLabels,
 	}, resp); err != nil {
```

Comment:
It seems like we were miscommunicating a bit. My focus was removing `resp` / `seriesServer` because it is an intermediary buffering step that buffers all of the responses however you are now focusing on removing `sortDedupLabels` which is very good :+1: we could remove `seriesServer` after